public/pages/Administration/pages/ManageAuthentication.page.tsx

```diff
@@ -55,69 +60,125 @@ export default class ManageAuthenticationPage extends AdminBasePage<ManageAuthen
     this.setState({ isAdding: false, editing: undefined })
   }
 
+  private toggleEmailAuth = async (active: boolean) => {
+    this.setState(
+      () => ({
+        isAllowingEmailAuth: active,
+      }),
+      async () => {
+        const response = await actions.updateTenantAllowingEmailAuth(this.state.isAllowingEmailAuth)
+        if (response.ok) {
+          notify.success(`You successfully ${this.state.isAllowingEmailAuth ? "allowed" : "disallowed"} email authentication.`)
+        } else {
+          this.setState(
+            () => ({
+              isAllowingEmailAuth: !active,
+              error: response.error,
+            }),
+            () => notify.error("Unable to save this setting.")
+          )
+        }
+      }
+    )
+  }
+
   public content() {
+    let enabledProvidersCount = 0
+    for (const o of this.props.providers) {
+      if (o.isEnabled) {
+        enabledProvidersCount++
+      }
+    }
+    const cantDisable = !this.state.isAllowingEmailAuth && enabledProvidersCount == 1
+
     if (this.state.isAdding) {
-      return <OAuthForm onCancel={this.cancel} />
+      return <OAuthForm cantDisable={cantDisable} onCancel={this.cancel} />
     }
 
     if (this.state.editing) {
-      return <OAuthForm config={this.state.editing} onCancel={this.cancel} />
+      return <OAuthForm cantDisable={cantDisable} config={this.state.editing} onCancel={this.cancel} />
     }
 
     const enabled = <span className="text-green-700">Enabled</span>
     const disabled = <span className="text-red-700">Disabled</span>
 
     return (
-      <>
-        <h2 className="text-display">OAuth Providers</h2>
-        <p>
-          You can use these section to add any authentication provider thats supports the OAuth2 protocol. Additional information is available in our{" "}
-          <a rel="noopener" className="text-link" target="_blank" href="https://getfider.com/docs/configuring-oauth/">
-            OAuth Documentation
-          </a>
-          .
-        </p>
-        <VStack spacing={6}>
-          {this.props.providers.map((o) => (
-            <div key={o.provider}>
-              <HStack justify="between">
-                <HStack className="h-6">
-                  <OAuthProviderLogo option={o} />
-                  <strong>{o.displayName}</strong>
+      <VStack spacing={8}>
+        <div>
+          <h2 className="text-display">General Authentication</h2>
+          <Form error={this.state.error}>
+            {this.state.isAllowingEmailAuth &&
+              (this.state.canDisableEmailAuth ? (
+                <p className="text-green-700">You can disable email auth because you have at least one authentication provider enabled</p>
```

Comment:
I think we simplify the messaging. This green text being displayed at all times is unnecessary in my option. I'm thinking of:

1. remove this green/red sub-message
2. as a helper text of the checkbox, if the input is disabled, explain why. If the checkbox is enabled, explain what happens when that is toggled off.

The note can be displayed at all times.

Reply:
Ok I agree the messaging wasn't optimal, is that better like this?
Please let me know if you have any comment about it now :blush: